panel/layout/feed.py

```diff
@@ -78,6 +78,11 @@ def __init__(self, *objects, **params):
         self._last_synced = None
         self.param.watch(self._trigger_view_latest, 'objects')
 
+    @param.depends("objects", watch=True)
+    def _reset_visible_range(self):
```

Comment:
This is **probably** overkill and maybe too late (not sure about the order of operations).

We likely need to:

1. Detect which items were added/removed, i.e. are all currently visible objects still present, likely the best thing would be to simply recompute the visible range based on the children that are still present.
2. We may need to do this inside `_update_model` so that we can adjust the visible_range before `_get_objects` recomputes the objects to send.